rocAL/rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -2534,3 +2552,143 @@ rocalResetLoaders(RocalContext p_context)
+    try {
+        if(shard_count < 1 )
+            THROW("Shard count should be bigger than 0")
+
+        if(shard_id >= shard_count)
+            THROW("Shard id should be smaller than shard count")
+        auto [max_frames, max_channels] = evaluate_audio_data_set(StorageType::FILE_SYSTEM, DecoderType::SNDFILE, source_path, "");
+        INFO("Internal buffer size for audio frames = "+ TOSTR(max_frames))
+        RocalTensorlayout tensor_layout = RocalTensorlayout::NONE;
+        RocalTensorDataType tensor_data_type = RocalTensorDataType::FP32;
+        std::vector<size_t> dims = {context->user_batch_size(), max_frames, max_channels};
+        auto info  = TensorInfo(std::vector<size_t>(std::move(dims)),
+                                     context->master_graph->mem_type(),
+                                     tensor_data_type);
+        info.set_max_shape();
+        info.set_tensor_layout(tensor_layout);
+        output = context->master_graph->create_loader_output_tensor(info);
+        output->reset_audio_sample_rate();
+        auto cpu_num_threads = context->master_graph->calculate_cpu_num_threads(shard_count);
+        context->master_graph->add_node<AudioLoaderSingleShardNode>({}, {output})->init(shard_id, shard_count, cpu_num_threads,
+                                                                                        source_path,
+                                                                                        source_file_list_path,
+                                                                                        StorageType(storage_type),
+                                                                                        DecoderType::SNDFILE,
+                                                                                        shuffle,
+                                                                                        loop,
+                                                                                        context->user_batch_size(),
+                                                                                        context->master_graph->mem_type(),
+                                                                                        context->master_graph->meta_data_reader()
+                                                                                        );
+        context->master_graph->set_loop(loop);
+
+        if(is_output)
+        {
+            auto actual_output = context->master_graph->create_tensor(info, is_output);
+            context->master_graph->add_node<CopyNode>({output}, {actual_output});
+        }
+
+    }
+    catch(const std::exception& e) {
+        context->capture_error(e.what());
+        std::cerr << e.what() << '\n';
+    }
+    return output;
+}
+
+RocalTensor  ROCAL_API_CALL
+rocalAudioFileSource(
+        RocalContext p_context,
+        const char* source_path,
+        const char* source_file_list_path,
```

Comment:
To remove this file_list_path now - will be added as a part of the file list reader (meta-data reader)

Reply:
done